support: unified process for handling CCPA/GDPR data deletion requests; add checklist issue template

emidoots opened the pull request:
[Someone asked in Slack](https://sourcegraph.slack.com/archives/C015P61EXV1/p1633973421077100) how we handle user data deletion requests. I couldn't find a clear page to direct them to, and it appears there are numerous systems/teams that this needs to occur in.

See the checklist sister PR: https://github.com/sourcegraph/customer/pull/519

I had previously built our GDPR-compliant data deletion functionality in the product itself (the 'nuke user account' button) and handled a fair number of these, although quite rare/infrequent, so I am trying to document our process for doing so in a single, unified place and am putting together an issue checklist in the customer repo to help this process go more smoothly.

@sourcegraph/cse @virginiaulrich Please take a look, these requests are very rare/infrequent (maybe a few  a year) and so I don't think it'll add any burden - should just make the process more clear - but I've put this in your handbook section.

@daxmc99 @bcreager I've marked Cloud DevOps as handling the actual nuking of user accounts on Sourcegraph.com, as this requires admin privileges. In practice I'll continue to handle these but just wanted to give a heads up. There is documentation here on how to do this as well.

Commit:
support: add a process for handling GDPR data deletion requests

[Someone asked](https://sourcegraph.slack.com/archives/C015P61EXV1/p1633973421077100) how we handle user data deletion requests.

I had previously built our GDPR-compliant data deletion functionality (the 'nuke user account' button) and handled a fair number of these, although quite rare/infrequent, so I am now documented our process for doing so.

slimsag committed:
squash! Prettier

Commit:
Link from index page

Commit:
ce: remove delete_users_guide.md (not handled by CE today, outdated page)

Commit:
Merge branch 'main' into sg/gdpr

tech-ops-admin commented:
Thank you for asking Stephen. While Tech Ops manage the access to HubSpot
and Salesforce, the data is managed by Sales Ops. Joe Kor Seth H can help
you here.

On Mon, Oct 11, 2021 at 1:12 PM 'Stephen Gutekanst' via Tech Ops <
***@***.***> wrote:

> Someone asked in Slack
> <https://sourcegraph.slack.com/archives/C015P61EXV1/p1633973421077100>
> how we handle user data deletion requests.
>
> I had previously built our GDPR-compliant data deletion functionality (the
> 'nuke user account' button) and handled a fair number of these, although
> quite rare/infrequent, so I am documenting our process for doing so and
> mapping it onto the modern teams we have set up.
>
> @sourcegraph/cse <https://github.com/orgs/sourcegraph/teams/cse>
> @virginiaulrich <https://github.com/virginiaulrich> Please take a look,
> these requests are very rare/infrequent (maybe a few a year) and so I don't
> think it'll add any burden - should just make the process more clear - but
> I've put this in your handbook section.
>
> @sourcegraph/tech-ops <https://github.com/orgs/sourcegraph/teams/tech-ops>
> I've marked you here for handling data deletion from both Hubspot and
> Salesforce. Are you the right team to own such requests, or should I e.g.
> reach out to the sales team about them doing this?
>
> @daxmc99 <https://github.com/daxmc99> @bcreager
> <https://github.com/bcreager> I've marked Cloud DevOps as handling the
> actual nuking of user accounts on Sourcegraph.com, as this requires admin
> privileges. In practice I'll continue to handle these but just wanted to
> give a heads up. There is documentation here on how to do this as well.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/sourcegraph/handbook/pull/250
> Commit Summary
>
>    - support: add a process for handling GDPR data deletion requests
>    <https://github.com/sourcegraph/handbook/pull/250/commits/cad91dfd55db2854ec44cd46937952e5244554a6>
>    - squash! Prettier
>    <https://github.com/sourcegraph/handbook/pull/250/commits/fa49b8252776255bb90b1177d9be5c19628ffb89>
>
> File Changes
>
>    - *A* content/support/permanently_deleting_user_data.md
>    <https://github.com/sourcegraph/handbook/pull/250/files#diff-8aa73f5f3e6b7e6a3b142e17d0ef10401d7d1be4e8f6413742ea381fd475d82c>
>    (51)
>
> Patch Links:
>
>    - https://github.com/sourcegraph/handbook/pull/250.patch
>    - https://github.com/sourcegraph/handbook/pull/250.diff
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/sourcegraph/handbook/pull/250>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AUTOGI3BRSVEXYA7GIAXKA3UGNAMTANCNFSM5FY7MHIQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>


-- 
*Nicky Van Maanen *(she/her)
Tech Ops Manager
[image: https://sourcegraph.com] <https://sourcegraph.com>

dadlerj commented:
I believe there are a number of systems beyond HubSpot/Salesforce that might require deletion. There are a few different potential sources to go to for this, list but at some point we need to make it concrete. Some relevant links:

* We used to have a page that covered GDPR compliance in the marketing operations part of the handbook at https://sourcegraph.com/github.com/sourcegraph/about@0f76214/-/blob/handbook/marketing/marketing_operations.md?L67:9 (see context for why this was deleted unintentionally as part of a larger refactor at https://sourcegraph.slack.com/archives/C01GJFE26BX/p1632416152223900?thread_ts=1632324726.206800&cid=C01GJFE26BX), but the details aren't very specific.
* https://handbook.sourcegraph.com/bizops/analytics#data-sources
* https://handbook.sourcegraph.com/bizops/customer_data_policy

I think this update is helpful for in-product/cloud data deletion, but being GDPR compliant requires a larger scale effort today, and I'd make that clear in this update if possible.

emidoots commented:
@dadlerj Yeah, there was also an outdated CE page for this I deleted. I'd like to gather all of these sources together to make it clear what our process here is (aside from.. spread across everywhere, as you know :) )

I will incorporate the things you linked into this page or link to them here as well, thanks!

Commit:
Replace #it-tech-ops with #sales-ops for Hubspot/Salesforce

daxmc99 reviewed (approved):
Approving from a DevOps standpoint 👍

emidoots commented:
I am going to create an issue template with checklist in the github.com/sourcegraph/customer repository for tracking the steps needed, since it spans so many teams/systems.

Commit:
reference checklist

emidoots commented:
I've updated this PR to primarily reflect the application-side aspects, and how e.g. Support can assist a customer in handling such requests on their own Sourcegraph instance.

I've moved all the other bits into a sourcegraph/customer repo issue template, which covers the current state of GDPR/CCPA handling (which requires some more white glove treatment) after discussing with Dan https://github.com/sourcegraph/customer/pull/519

@dadlerj I also moved the handbook bit you linked to which no longer existed in the handbook, into the issue template.

Commit:
Update ce/education index

virginiaulrich commented on content/support/permanently_deleting_user_data.md:
```diff
@@ -0,0 +1,36 @@
+
+### In the product: how to nuke a user account
+
+If the request is for a Sourcegraph.com account, a #dev-ops team member will handle this step for you (it requires site admin).
```
@slimsag this is not correct. We maintain a small subset of the team with .com access so they can do this without having to engage engineering. I will update this to reflect that.

emidoots replied:
Oh perfect, even better!

virginiaulrich commented on content/support/permanently_deleting_user_data.md:
```diff
@@ -0,0 +1,36 @@
+
+The support team member is responsible for delegating to other teams. If you're not the support team, you should inform the user that only our support team support@sourcegraph.com can handle such requests.
+
+[Create an issue in the sourcegraph/customer repository](https://github.com/sourcegraph/customer/issues/new/choose) and follow the "**Customer support: Delete user data**" checklist issue template.
```
I think I am missing where I can review this template to see what all is required of CS. Can you help me understand where the full list is?

emidoots replied:
Oh yes, sorry, it's here: https://github.com/sourcegraph/customer/pull/519

virginiaulrich committed:
Clarifying that some CSEs do have admin access

virginiaulrich reviewed (approved):
Thanks, Stephan! Should be all set now that I've updated that we have a few folks with ability to nuke user accounts.

Commit:
Merge branch 'main' into sg/gdpr

emidoots commented:
Merging since there's no more feedback, we can always iterate on this further.
